### PR TITLE
docs(catboost): improve CatBoost documentation

### PR DESCRIPTION
**Pull request by sauyon**
Rewrote a bunch of docs for clarity. Should happen for the rest of them but I thought I'd put this up now for review.

@timliubentoml maybe to take a look?

### CODE REVIEW

**Comment by sauyon** [comment]
docs preview: https://bentoml--2359.org.readthedocs.build/en/2359/

File: bentoml/_internal/frameworks/catboost.py [hunk 1]
```diff
@@ -114,51 +108,45 @@ def load(
...
+            A dict of immutable default labels for use in model-management programs such as Yatai, e.g. team=nlp, stage=dev
+        custom_objects (``dict[str, Any]``, `optional`, default ``None``):
+            Additional python objects to be saved alongside the model,
             e.g. a tokenizer instance, preprocessor function, model configuration json
-        metadata (:code:`Dict[str, Any]`, `optional`,  default to :code:`None`):
-            Custom metadata for given model.
-        model_store (:mod:`~bentoml._internal.models.store.ModelStore`, default to :mod:`BentoMLContainer.model_store`):
-            BentoML modelstore, provided by DI Container.
+        metadata (``dict[str, Any]``, `optional`,  default ``None``):
+            Custom metadata for given model. TODO: write metadata documentation.
```
Comment (parano) [inline bentoml/_internal/frameworks/catboost.py:146]: For metadata, let's use the same description in docstring from `Model.create`?

```
            name: model name in target model store, model version will be automatically
                generated
            module: import path of module used for saving/loading this model, e.g.
                "bentoml.tensorflow"
            labels:  user-defined labels for managing models, e.g. team=nlp, stage=dev
            options: default options for loading this model, defined by runner
                implementation, e.g. xgboost booster_params
            custom_objects: user-defined additional python objects to be saved
                alongside the model, e.g. a tokenizer instance, preprocessor function,
                model configuration json
            metadata: user-defined metadata for storing model training context
                information or model evaluation metrics, e.g. dataset version,
                training parameters, model scores
            context: Environment context managed by BentoML for loading model,
                e.g. {"framework:" "tensorflow", "framework_version": _tf_version}
```

File: bentoml/_internal/frameworks/catboost.py [hunk 2]
```diff
@@ -114,51 +108,45 @@ def load(
...
-        model_export_parameters (:code:`Dict[str, Union[str, Any]]`, `optional`, default to :code:`None`):
-            Export parameters for given model.
-        model_pool (:code:`cbt.core.Pool`, `optional`, default to :code:`None`):
-            CatBoost data pool for given model.
-        labels (:code:`Dict[str, str]`, `optional`, default to :code:`None`):
-            user-defined labels for managing models, e.g. team=nlp, stage=dev
-        custom_objects (:code:`Dict[str, Any]]`, `optional`, default to :code:`None`):
-            user-defined additional python objects to be saved alongside the model,
+        name (``str``):
+            The name for given model instance. This must be a valid :obj:`~bentoml._internal.types.Tag` name; see
```
Comment (larme) [inline bentoml/_internal/frameworks/catboost.py:131]: I think there's the links to doc of `Tag` after "see"?

Comment (larme) [inline bentoml/_internal/frameworks/catboost.py:131]: I think there's the links to doc of `Tag` after "see"

File: bentoml/_internal/frameworks/catboost.py [hunk 3]
```diff
@@ -114,51 +108,45 @@ def load(
...
-        metadata (:code:`Dict[str, Any]`, `optional`,  default to :code:`None`):
-            Custom metadata for given model.
-        model_store (:mod:`~bentoml._internal.models.store.ModelStore`, default to :mod:`BentoMLContainer.model_store`):
-            BentoML modelstore, provided by DI Container.
+        metadata (``dict[str, Any]``, `optional`,  default ``None``):
+            Custom metadata for given model. TODO: write metadata documentation.
 
     Returns:
-        :obj:`~bentoml._internal.types.Tag`: A :obj:`tag` with a format `name:version` where `name` is the user-defined model's name, and a generated `version` by BentoML.
+        :obj:`~bentoml._internal.types.Tag`: A tag that can be used to access the model from the BentoML model store.
```
[the remark this answers is not in the document]
Comment (sauyon) [reply bentoml/_internal/frameworks/catboost.py:149]: The reason I chose to remove it was I think that it's not really relevant information to the user at this point.

File: bentoml/_internal/frameworks/catboost.py (hunk 2 above)
Comment (sauyon) [reply bentoml/_internal/frameworks/catboost.py:131]: Going to just remove the 'see' part until we have real `Tag` docs.

**Review by larme — APPROVED** [review]

**Pull request closed without merging**